File: for-doc-java/src/main/java/org/doc/j/model/BookStore.java
package org.doc.j.model;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@Entity
public interface BookStore {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    String name();

    @Nullable
    String website();

    @OneToMany(mappedBy = "store")
    List<Book> books();
}

File: for-doc-kotlin/src/main/kotlin/org/doc/k/model/BookStore.kt
package org.doc.k.model

import org.babyfish.jimmer.sql.*

@Entity
interface BookStore {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    val name: String

    val website: String?

    @OneToMany(mappedBy = "store")
    val books: List<Book>
}